### PR TITLE
Ndlstm dynamic batch size

### PR DESCRIPTION
**Pull request by jvgonzalvo**
Apparently, lstm2d.separable_lstm doesn't accept dynamic batch sizes (number of images). This pull request addresses issue https://github.com/tensorflow/tensorflow/issues/16510

### CODE REVIEW

**Comment by selcouthlyBlue** [comment]
I actually tried it for my own project. The loss was not really affected that much.

**Comment by selcouthlyBlue** [comment]
I tried it for my own project and it's taking long to get to a step in training. Any other ideas on how to make the batch size for ndlstm dynamic?

**Comment by selcouthlyBlue** [comment]
It's now working as intended. Test please.

**Comment by selcouthlyBlue** [comment]
I forgot to revert the changes I made in lstm2d_test.py that was working for my previous implementation of lstm2d.py. My bad. Test again please.

**Comment by selcouthlyBlue** [comment]
I wasn't able to run tests in my own machine before so I couldn't really figure out if it works or not. I managed to run the tests now and it's working perfectly.

**Comment by selcouthlyBlue** [comment]
I got an [error](https://source.cloud.google.com/results/invocations/c4cd9871-91f0-4a4b-84d6-cf3918b1e7ea/targets/%2F%2Ftensorflow%2Fcontrib%2Fbayesflow:hmc_test/log) that seems to be totally unrelated to the module I'm working on.

**Comment by drpngx** [comment]
please resolve conflicts

**Comment by selcouthlyBlue** [comment]
I'm closing this pull request now since ndlstm is removed from tensorflow:master.
